cmd/containerd-shim-runc-v2/task/service.go

```diff
@@ -299,9 +300,15 @@ func (s *service) Start(ctx context.Context, r *taskAPI.StartRequest) (*taskAPI.
 	if err != nil {
 		return nil, err
 	}
-
 	var cinit *runc.Container
 	s.lifecycleMu.Lock()
+	if s.containerInitExited != nil {
+		exited, ok := s.containerInitExited[container.ID]
```

Comment:
I don't think we need to track this state explicitly. It's already available from `container`.
```go
initp, err := container.Process("")
if err != nil {
    return nil, err
}
containerInitExited := !initp.ExitedAt().IsZero()
```

Reply:
Nope, I'm wrong. `(process.Process).ExitedAt()` is set when `(process.Process).SetExited()` is called, which the task service does in `handleProcessExits()`... _after_ killing all execs. That also neuters my idea of having the `container.Start()` call return an error if the exec's parent has exited.

Reply:
init exited event doesn't have chance run handleProcessExit.(because it have to  run at last )
```
func (s *service) handleProcessExit(e runcC.Exit, c *runc.Container, p process.Process) {
	if ip, ok := p.(*process.Init); ok {
		// Ensure all children are killed
		if runc.ShouldKillAllOnExit(s.context, c.Bundle) {
			if err := ip.KillAll(s.context); err != nil {
				log.G(s.context).WithError(err).WithField("id", ip.ID()).
					Error("failed to kill init's children")
			}
		}
	}

	p.SetExited(e.Status)
	s.send(&eventstypes.TaskExit{
		ContainerID: c.ID,
		ID:          p.ID(),
		Pid:         uint32(e.Pid),
		ExitStatus:  uint32(e.Status),
		ExitedAt:    protobuf.ToTimestamp(p.ExitedAt()),
	})
}
```